build.sh

```diff
@@ -1,117 +1,123 @@
-#!/usr/bin/env bash
-
-##########################################################################
-# This is the Cake bootstrapper script for Linux and OS X.
-# This file was downloaded from https://github.com/cake-build/resources
-# Feel free to change this file to fit your needs.
-##########################################################################
-
-# Define directories.
-SCRIPT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
-TOOLS_DIR=$SCRIPT_DIR/tools
-ADDINS_DIR=$TOOLS_DIR/Addins
-MODULES_DIR=$TOOLS_DIR/Modules
-NUGET_EXE=$TOOLS_DIR/nuget.exe
-CAKE_EXE=$TOOLS_DIR/Cake/Cake.exe
-PACKAGES_CONFIG=$TOOLS_DIR/packages.config
-PACKAGES_CONFIG_MD5=$TOOLS_DIR/packages.config.md5sum
-ADDINS_PACKAGES_CONFIG=$ADDINS_DIR/packages.config
-MODULES_PACKAGES_CONFIG=$MODULES_DIR/packages.config
-
-# Define md5sum or md5 depending on Linux/OSX
-MD5_EXE=
-if [[ "$(uname -s)" == "Darwin" ]]; then
-    MD5_EXE="md5 -r"
+#!/bin/bash
+
+set -euo pipefail
+
+OS="$(uname)"
+
+# Set up environment variables based on OS
+if [[ "$OS" == "Darwin" ]]; then
+  export PKG_CONFIG_PATH="/usr/local/lib/pkgconfig:/opt/homebrew/lib/pkgconfig"
+  export CXXFLAGS="-std=c++17 -fPIC -O3 -g -I/opt/homebrew/include -I/usr/local/include"
+  export LDFLAGS="-L/opt/homebrew/lib -L/usr/local/lib"
+  DYLIB_EXT="dylib"
+  DYLIB_PREFIX="lib"
+  RID="osx-x64"
```

Comment:
I suspect your CMake Lists just buidls M series TBH. I also don't think we use this RID var anywhere in this script now. 

Perhaps we _should_ have a tripples or similar so we can choose to build both on MacOS? Or just leae it building AArch64 and be done with it. Most devs will be using M series these days.

Reply:
Added